src/main/java/courseManager/Assessment.java

```diff
@@ -0,0 +1,70 @@
+package courseManager;
+
+import weightScheme.WeightScheme;
+
+import java.util.ArrayList;
+
+public class Assessment {
+    private String title;
+    private InstanceList instanceList;
+    private WeightScheme weightScheme;
+
+    /**
+     * Create a new Assessment object with a title and an associated WeightScheme
+     * @param title The title of the assessment
+     *              e.g. "Quizzes", "Midterm", "Homework Assignments"
+     * @param weightScheme The WeightScheme associated with the assessment
+     *              e.g. "Quizzes" might be associated with a SimpleWeight scheme with a weight of 10% for each quiz
+     */
+
+    public Assessment(String title, WeightScheme weightScheme) {
+        this.title = title;
+        this.weightScheme = weightScheme;
+        this.instanceList = new InstanceList(this.title, weightScheme.getNumberOfInstances());
+
+    }
+
+    public float getTotalWeight() {
+        float totalWeight = 0.0f;
+        /*iterate through each instance in the instance list and add its weight to the total weight*/
+        return totalWeight;
+    }
+
+    public float getCommittedWeight() {
+        float committedWeight = 0.0f;
+        /*iterate through the instance list and add committed instances' weight*/
```

Comment:
No need to iterate. Just grab the list of committed marks from the InstanceList and feed them to the "computeWeighted" function in the weightscheme.

Reply:
the issue is that this function shouldn't involve anything about marks, but only the weight of each instance that is committed. So far it is not even used anywhere in my package but I included it if anyone else would need it